quickwit-cli/src/indexing/document_indexer.rs

```diff
@@ -0,0 +1,130 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use crate::cli_command::IndexDataArgs;
+use tantivy::{schema::Schema, Document};
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio::task;
+use tokio::task::JoinHandle;
+
+use crate::indexing::statistics::StatisticEvent;
+
+use crate::indexing::split::Split;
+
+pub async fn index_documents(
+    args: &IndexDataArgs,
+    mut document_receiver: Receiver<String>,
+    statistic_sender: Sender<StatisticEvent>,
```

Comment:
I like that you thought of a way to collect statistics!
I wonder if a channel is the right way to do this, but we can keep it that way for the moment.

Reply:
The reason I use channel is that the spec requested to display the throughput as the work is in progress. so this kind of is like an event-based statistic collection.

Reply:
Having statistics is a good idea. Using is not necessarily the best way.